tools/codegen/gen.py

```diff
@@ -477,24 +441,13 @@ def go(f: NativeFunction) -> Optional[str]:
         dispatcher_exprs = dispatcher.legacydispatcherarguments_exprs(legacy_dispatcher_args)
 
         if target is Target.DEFINITION:
-            # See Note [Byte-for-byte compatibility]
-            # I don't think there's actually a good reason to generate
-            # these two cases differently
-            if legacy_dispatcher_tensor_args:
```

Comment:
It's not actually clear to me why these two cases are okay to merge together, since their logic looks pretty different. The first path does a few additional things:
- includes dispatch keys from any tensor argument, removing keys with a higher priority than backend select
- includes dispatch keys from TLS (which I don't believe computeDispatchKey() does)

I could imagine that the input tensors to factory functions aren't supposed to have dispatch keys on them (?) resolving point 1, but I'm not sure about point 2